src/Microsoft.AspNetCore.Razor.Evolution/DefaultRazorRuntimeCSharpLoweringPhase.cs

```diff
@@ -34,6 +34,13 @@ protected override void ExecuteCore(RazorCodeDocument codeDocument)
                 LineMappings = renderingContext.Writer.LineMappingManager.Mappings,
             };
 
+            if (renderingContext.ErrorSink.Errors.Count > 0)
+            {
+                var combinedErrors = syntaxTree.Diagnostics.Concat(renderingContext.ErrorSink.Errors).ToList();
+                syntaxTree = RazorSyntaxTree.Create(syntaxTree.Root, combinedErrors, syntaxTree.Options);
+            }
+
+            codeDocument.SetSyntaxTree(syntaxTree);
```

Comment:
We shouldn't be adding errors to the syntax tree in a later phase. There errors belong on the CSharp document